### PR TITLE
 Move stdlib deprecations to @deprecated attribute

### PR DESCRIPTION
**Pull request by josevalim**

### CODE REVIEW

**Comment by michalmuskala** [comment]
What if we had a pass that collected all the `@deprecated` attributes from stdlib and baked them into the deprecation module? 

Also, if we're going to be loading the modules anyway, why not do something like this:

```erl
get_deprecations(Module) ->
  try Module:'__info__'(deprecations)
  catch _:_ -> []
  end.
```

**Comment by josevalim** [comment]
@michalmuskala i will benchmark but, if i remember correctly, the false positive cost in this case is very high. And it may happen frequently, especially when calling erlang modules, so we may need to explicitly write those down.

**Comment by josevalim** [comment]
@ericmj let's please not mix discussions.

**Comment by josevalim** [comment]
@ericmj regarding making it a map, it could speed things up a bit but I don't think it will make a different for entries with a handful of elements.

**Comment by josevalim** [comment]
@michalmuskala here are the benchmark results for 10000 calls:

```
===loaded+exported+try===
unknown module: 18059130us
erlang module: 1316us
elixir module: 1536us
===try===
unknown module: 17983240us
erlang module: 42838us
elixir module: 861us
```

You can see it makes the elixir module almost twice faster but it makes the erlang ones 32 times slower.

**Comment by josevalim** [comment]
I could inline both Kernel and erlang modules, it makes their respective lookups 3 times faster, but note we are talking about 10000 calls going from 1500us to 500us.

**Comment by whatyouhide** [comment]
@josevalim nitpick, but for some messages you're saying "Use x instead" and for other messages you're saying "Use x". Maybe make them uniform?

**Comment by whatyouhide** [comment]
@josevalim I prefer the "instead" version but only slightly, doesn't really matter. Just pick one and stick to that :)

**Comment by josevalim** [comment]
I have added "instead". I think this is good to go now!

File: lib/elixir/lib/registry.ex [hunk 1]
```diff
@@ -325,9 +325,7 @@ defmodule Registry do
   end
 
   @doc false
-  @since "1.4.0"
-  @spec start_link(keys, registry, keyword) :: {:ok, pid} | {:error, term}
-  # TODO: Deprecate old start_link registry API on v1.8
+  @deprecated "Use Registry.start_link/1 instead"
```
[the remark this answers is not in the document]
Comment (josevalim) [reply lib/elixir/lib/registry.ex]: It was already deprecated.

**Comment by josevalim** [comment]
:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:

**Pull request merged**